### PR TITLE
Added Rack-CORS and it's settings

### PR DESCRIPTION
**Pull request by vladimirbazhanov**
### Summary

Added Rack-CORS and it's settings

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -33,5 +33,12 @@ class Application < Rails::Application
     # Middleware like session, flash, cookies can be added back manually.
     # Skip views, helpers and assets when generating a new resource.
     config.api_only = true
+
+    config.middleware.insert_before 0, Rack::Cors do
+      allow do
+        origins "localhost:5000", "lvh.me:5000", "127.0.0.1:5000"
```
Comment (ArthurZaharov) [inline config/application.rb:39]: Let's use ENV variables instead of hardcoded values.

Comment (ArthurZaharov) [reply config/application.rb:39]:
```
origins *ENV.fetch("ALLOW_CORS_REQUESTS_FROM", "")
```

File: config/application.rb [hunk 2]
```diff
@@ -33,5 +33,12 @@ class Application < Rails::Application
...
+        resource "*", headers: :any, methods: [:get, :post, :put, :patch, :delete]
```
Comment (ArthurZaharov) [inline config/application.rb:40]: They send OPTIONS request too. I think that we can open all methods for CORS
```suggestion
        resource "*", headers: :any, methods: :any
```

File: config/application.rb [hunk 3]
```diff
@@ -33,5 +33,14 @@ class Application < Rails::Application
     # Middleware like session, flash, cookies can be added back manually.
     # Skip views, helpers and assets when generating a new resource.
     config.api_only = true
+
+    allowed_hosts = ENV.fetch("ALLOW_CORS_REQUESTS_FROM", "").split(",")
+
+    config.middleware.insert_before 0, Rack::Cors do
```
Comment (ArthurZaharov) [inline config/application.rb:39]: Rails recommend to [use initializers](https://github.com/fs/rails-base-graphql-api/pull/11/files#diff-b1fe55db50c712fef0673345e5b9c0d9R28).
```
Application configuration can go into files in config/initializers
```

Can we extract it to config/iniitalizers/rack_cors.rb

File: config/application.rb [hunk 4]
```diff
@@ -33,5 +33,14 @@ class Application < Rails::Application
     # Middleware like session, flash, cookies can be added back manually.
     # Skip views, helpers and assets when generating a new resource.
     config.api_only = true
+
+    allowed_hosts = ENV.fetch("ALLOW_CORS_REQUESTS_FROM", "").split(",")
```
Comment (timurvafin) [inline config/application.rb:37]: These settings could be moved to `https://github.com/fs/rails-base-api/blob/master/config/initializers/cors.rb`

**Pull request merged**